File: lib/ios/native/Midi/MidiPlayer.m
//
//  MidiPlayer.m
//  horizontalPicker
//
//  Created by Valentin Blokhin on 22.03.2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

#import "React/RCTBridgeModule.h"


@interface RCT_EXTERN_MODULE(MidiPlayer, NSObject)

RCT_EXTERN_METHOD(load: (NSString)path
                  resolve: (RCTPromiseResolveBlock)resolve
                  reject: (RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(playMidiNote: (NSInteger)midi)
RCT_EXTERN_METHOD(playMidiNotes: (NSArray)midis)
RCT_EXTERN_METHOD(stopMidiNote: (NSInteger)midi)
RCT_EXTERN_METHOD(stopMidiNotes: (NSArray)midis)

RCT_EXTERN_METHOD(unmute)
@end

File: lib/ios/native/Midi/MidiPlayer.swift
//
//  MidiPlayer.swift
//  horizontalPicker
//
//  Created by Valentin Blokhin on 22.03.2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation
import AVFoundation

@objc(MidiPlayer)
class MidiPlayer: NSObject {
    
    var au: AudioUnitMIDISynth!
    
    @objc
    func load(_ path: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
        let url = URL(fileURLWithPath: path)
        au = AudioUnitMIDISynth(soundfont: url)
        
        print("Valid URL: \(url)")
        let message = "Prepared Sound Font"
        print(message)
        
        resolve(message)
    }
    
  @objc
    func playMidiNote(_ midi: Int) {
        au.playPitch(midi: midi)
  }
  @objc
    func stopMidiNote(_ midi: Int) {
    au.stopPitch(midi: midi)
  }
    
    
    @objc
      func playMidiNotes(_ midis: [Int]) {
        for midi in midis {
            au.playPitch(midi: midi)
        }
    }
    
    @objc
      func stopMidiNotes(_ midis: [Int]) {
        for midi in midis {
            au.stopPitch(midi: midi)
        }
    }
    
    @objc
    func unmute() {
        do {
            try AVAudioSession.sharedInstance().setCategory(AVAudioSession.Category.playback)
        } catch {
            print(error)
        }
    }
    }
